Bottom: 946b716fc954fca34fa7197567a22429090c7b28
Top:    d28e13fedbcbab8b6de4e9313a32c483a52bdccd
Author: Ian Kent <raven@themaw.net>
Date:   2013-05-27 12:49:49 +0800

autofs-5.0.7 - dont prode rdma mounts

Since the change to pass text nfs mount options drectly to the kernel all autofs
mount requests now probe server availability. This was because of long delays
mounting from servers that aren't responding.

This caused mounts requesting the rdma protocol to fail if udp or tcp was also
not available from the server.

Since, AFAICT the rmda protocol can't be used with RPC fromn userspace, the only
way to work around it is to not probe servers when rdma is requested.



---

diff --git a/modules/mount_nfs.c b/modules/mount_nfs.c
index e61320b..f3718d5 100644
--- a/modules/mount_nfs.c
+++ b/modules/mount_nfs.c
@@ -71,6 +71,7 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 	int nosymlink = 0;
 	int port = -1;
 	int ro = 0;            /* Set if mount bind should be read-only */
+	int rdma = 0;
 
 	if (ap->flags & MOUNT_FLAG_REMOUNT)
 		return 0;
@@ -136,6 +137,9 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 				flags &= ~MOUNT_FLAG_USE_WEIGHT_ONLY;
 			} else if (strncmp("use-weight-only", cp, o_len) == 0) {
 				flags |= MOUNT_FLAG_USE_WEIGHT_ONLY;
+			} else if (strcmp("proto=rdma", cp, o_len) == 0 ||
+				   strcmp("rdma", cp, o_len) == 0) {
+				rdma = 1;
 			} else {
 				if (strncmp("vers=4", cp, o_len) == 0 ||
 				    strncmp("nfsvers=4", cp, o_len) == 0)
@@ -170,7 +174,12 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 		info(ap->logopt, MODPREFIX "no hosts available");
 		return 1;
 	}
-	prune_host_list(ap->logopt, &hosts, vers, port);
+	/*
+	 * We can't probe protocol rdma so leave it to mount and
+	 * and suffer the delay if a server isn't available.
+	 */
+	if (!rdma)
+		prune_host_list(ap->logopt, &hosts, vers, port);
 
 	if (!hosts) {
 		info(ap->logopt, MODPREFIX "no hosts available");
